pkg/ccl/streamingccl/streamingest/alter_replication_job.go

```diff
@@ -198,6 +219,91 @@ func alterReplicationJobHook(
 		if err != nil {
 			return err
 		}
+
+		// If a source address is being provided, we're enabling replication into an
+		// existing virtual cluster. It must be inactive, and we'll verify that it
+		// was the cluster from which the one it will replicate was replicated, i.e.
+		// that we're reversing the direction of replication. We will then revert it
+		// to the time they diverged and pick up from there.
+		if alterTenantStmt.ReplicationSourceAddress != nil {
+			dstTenantID, err := roachpb.MakeTenantID(tenInfo.ID)
```

Comment:
nit: for easier readability, could you pull this branch's logic into its own helper function?

Reply:
Done.